node/tessel-export.js

```diff
@@ -640,6 +719,29 @@ Tessel.Pin.prototype.analogWrite = function(val) {
   return this;
 };
 
+// Duty cycle should be a value between 0 and 1
+Tessel.Pin.prototype.pwmDutyCycle = function(dutyCycle, cb) {
+  // throw an error if this pin doesn't support PWM
+  if (!this.pwmSupported) {
+    throw new Error('PWM can only be used on G1 and G2 of either module port.');
+  }
+
+  if (typeof dutyCycle !== 'number' || dutyCycle > 1.0 || dutyCycle < 0) {
+    throw new Error('PWM duty cycle must be a number between 0 and 1');
+  }
+
+  // The frequency must be set prior to setting the duty cycle
+  if (Tessel.pwmPeriod === 0) {
+    throw new Error('PWM Frequency is not configured. You must call Tessel.pwmFrequency before setting duty cycle.');
+  }
+
+  var dutyCycleBuffer = new Buffer(2);
+  dutyCycleBuffer.writeUInt16BE(Math.floor(dutyCycle * Tessel.pwmPeriod));
+
+  this._port.sock.write(Buffer.concat([new Buffer([CMD.PWM_DUTY_CYCLE, this.pin]), dutyCycleBuffer]), cb);
+  return this;
+};
```

Comment:
:+1: Point unequivocally proven! Not only is your solution faster, but it's easier to read that what I was prototyping with. 

Also, I hadn't seen `hrtime` before so thanks for teaching me about it.